test/resources/components/component-with-name-prop.vue

```diff
@@ -0,0 +1,12 @@
+<template>
+  <div>
```

Comment:
Should this be a functional component? Since the patch is related to functional component name?

Reply:
No, the patch is related to functional components but the fix is for normal components.

Eg for a functional component when you access `vm.name` you will get the name of the component. But accessing `vm.name` for a normal component will return the name property _if_ that exists. Previously the code was just too greedy by always preferring to use `vm.name` if it exists. But given the aformentioned that will result in a find bug when you try to find a normal component that has a name prop itself